src/rust/src/lazy/dataframe.rs

```diff
@@ -327,9 +327,15 @@ impl LazyFrame {
         let expr_vec = robj_to!(VecPLExprCol, exprs)?;
         let maintain_order = robj_to!(Option, bool, maintain_order)?.unwrap_or(false);
         if maintain_order {
-            Ok(LazyGroupBy(self.0.clone().group_by_stable(expr_vec)))
+            Ok(LazyGroupBy {
+                lgb: self.0.clone().group_by_stable(expr_vec),
+                opt_state: self.0.get_current_optimizations(),
+            })
         } else {
-            Ok(LazyGroupBy(self.0.clone().group_by(expr_vec)))
+            Ok(LazyGroupBy {
+                lgb: self.0.clone().group_by(expr_vec),
+                opt_state: self.0.get_current_optimizations(),
+            })
```

Comment:
The current implementation is clearly not ideal.
Perhaps it would be better to convert to Rust's LazyGroupBy just before the aggregate function, like GroupBy on the R side.

Reply:
Thinking about it again, it's difficult considering the existence of things like `group_by_dynamic`...